src/tools/cephfs/cephfs-shell

```diff
@@ -1124,23 +1124,33 @@ sub-directories, files')
         """
         Disk Usage of a Directory
         """
-        if args.dirs == '':
-            args.dirs = cephfs.getcwd().decode('utf-8')
-        for dir_ in args.dirs:
-            if args.r:
-                for i in reversed(sorted(set(dirwalk(dir_)))):
-                    i = os.path.normpath(i)
-                    try:
-                        xattr = cephfs.getxattr(to_bytes(i), 'ceph.dir.rbytes')
-                        self.poutput('{:10s} {}'.format(
-                            humansize(int(xattr.decode('utf-8'))), '.' + i))
-                    except libcephfs.Error:
-                        continue
-            else:
-                dir_ = os.path.normpath(dir_)
-                self.poutput('{:10s} {}'.format(humansize(int(cephfs.getxattr(
-                    to_bytes(dir_), 'ceph.dir.rbytes').decode('utf-8'))), '.'
-                    + dir_))
+        dir_passed = args.dirs
+        if args.dirs is '':
+            dir_passed = cephfs.getcwd().decode('utf-8')
+            if len(dir_passed) > 1:
+                dir_passed = dir_passed.split()
+
+        for dir_ in dir_passed:
+            try:
+                dir_trav = sorted(set(dirwalk(dir_)))
+                if args.dirs and not dir_trav:
+                    dir_trav.append(dir_)
+                else:
+                    dir_trav.append('.')
+
+                if args.r:
+                    dir_trav = reversed(dir_trav)
+
+                for i in dir_trav:
+                    i_path = os.path.normpath(i)
+                    if (i is not '.') and (i_path[0] is '/'):
+                        i = '.' + i_path
+
+                    xattr = cephfs.getxattr(to_bytes(i_path), 'ceph.dir.rbytes')
+                    self.poutput('{:10s} {}'.format(humansize(int(xattr.decode(
+                        'utf-8'))), i))
+            except libcephfs.Error:
+                self.perror('{}: no such directory exists'.format(dir_), False)
```

Comment:
We do need `continue` but with the try-except inside the inner for loop. With current code, `du` won't continue after an error. This would be different from last time's pattern (i.e. `continue` outside for loop), since `continue` was ineffective

Reply:
Thank you for the clarification. I have modified the code.